src/main/java/htsjdk/samtools/cram/structure/SubstitutionMatrix.java

```diff
@@ -20,160 +20,251 @@
 import java.util.Arrays;
 import java.util.Comparator;
 
+/**
+ * Substitution matrix, used to represent base substitutions for reference-based CRAM
+ * compression.
+ *
+ * The matrix is stored internally in two forms; the packed/encoded form used for serialization,
+ * and an expanded in-memory form used for fast bi-directional interconversion between bases and
+ * substitution codes during read and write.
+ *
+ * This implementation substitutes both upper and lower case versions of a give base with the
+ * same (upper case) substitute base.
+ */
 public class SubstitutionMatrix {
-    public static final byte[] BASES = new byte[]{'A', 'C', 'G', 'T', 'N'};
-    private static final byte[] BASES_LC = new byte[]{'a', 'c', 'g', 't', 'n'};
-    private static final byte[] ORDER;
+    // List of bases subject to substitution. The upper and lower case arrays must have the bases
+    // in the same order so that the index for a given base is the same for upper and lower case
+    public static final byte[] SUBSTITUTION_BASES_UPPER = new byte[]{'A', 'C', 'G', 'T', 'N'};
```

Comment:
This continues to be unsafe and may be rewritten by the caller.  Is there a strong reason that this one is public and the others aren't?

Reply:
Maybe make this package protected if it needs to be used by tests.

Reply:
Yes, it should be package protected.